### DISCUSSION TITLE
How to access mocked constructor arguments in `answers`?

### DISCUSSION THREAD

**Discussion by elect86** [mockk/mockk#1223] [Q&A]
I have the following class

`class APIcreateBranchRequest (val repository: String, val branchCreation: BranchCreation)`

I'd like to capture somehow the constructor arguments to use them in `answers` lambda

```
        every { constructedWith<APIcreateBranchRequest>().execute() } answers {
            // `repository` and `branchCreation` are not available
            assert(repository == branch.repoId)
            assert(branchCreation.name == branch.id)
            assert(branchCreation.source == source)
            source
        }
```

how can I do that?

Ps: I found [this](https://github.com/mockk/mockk/issues/875) but the guy doesn't use `every` properly
Pps: expanded question can be found [here](https://stackoverflow.com/questions/78068110/how-to-port-a-python-monkeypatch-in-kotlin)

**Comment by Raibaz** [comment]
I'm personally not a big fan of running assertions in `answers` blocks: IMHO, they should be used when you need to run some logic to compute the value to be returned by a mocked method, but I prefer keeping stubbing (i.e., the "when" of my tests) separated from assertions (i.e. the "then" of my tests)

This being said, I think you can achieve what you need with something like this:

```
val repositorySlot = slot<String>()
val branchCreationSlot = slot<BranchCreation>()
every { 
     constructedWith<APICreateBranchRequest>(capture(repositorySlot), capture(branchCreationSlot)).execute()
} returns source

assertEquals(repositorySlot.captured, branch.repoId)
assertEquals(branchCreationSlot.captured.name, branch.id)
```
